### PR TITLE
Add validation EXT_API

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Also, add tests for the validation function by calling it through the new EXT_API.

### CODE REVIEW

File: samples/bootloader/prj.conf [hunk 1]
```diff
@@ -15,15 +15,17 @@ CONFIG_SYS_CLOCK_EXISTS=n
 CONFIG_NRF_RTC_TIMER=n
 CONFIG_CLOCK_CONTROL=n
 CONFIG_FPROTECT=y
+CONFIG_FW_INFO=y
```
Comment (hakonfam) [inline samples/bootloader/prj.conf]: Why does this need to be set explicitly?

File: subsys/bootloader/Kconfig [hunk 2]
```diff
@@ -26,7 +26,7 @@ config SECURE_BOOT_CRYPTO_CLIENT
 config SECURE_BOOT
 	bool "Use Secure Bootloader"
 	depends on !IS_SECURE_BOOTLOADER
-	select SECURE_BOOT_VALIDATION
+	select FW_INFO
```
Comment (hakonfam) [inline subsys/bootloader/Kconfig]: depends on?

Comment (hakonfam) [reply subsys/bootloader/Kconfig]: Why do we need this select now, how was this solved before? Did the bootloader cmake code inject the kconfig to the application?

File: subsys/bootloader/bl_validation/Kconfig [hunk 3]
```diff
@@ -4,13 +4,11 @@
...
-	bool
-	prompt "Secure Boot firmware validation"
-	select SECURE_BOOT_CRYPTO
-	select FW_INFO
+menu "Secure Boot firmware validation"
 
-if SECURE_BOOT_VALIDATION
+config SECURE_BOOT_VALIDATION
+	bool "Enable Secure Boot validation code"
+	select FW_INFO
```
Comment (hakonfam) [inline subsys/bootloader/bl_validation/Kconfig]: This is selected from both here and samples/bootloader. Should this be depends?

File: subsys/bootloader/Kconfig (hunk 2 above)
Comment (oyvindronningstad) [reply subsys/bootloader/Kconfig]: Changed to `depends on`

File: subsys/bootloader/bl_validation/Kconfig (hunk 3 above)
Comment (oyvindronningstad) [reply subsys/bootloader/bl_validation/Kconfig]: Changed to `depends on`

File: samples/bootloader/prj.conf (hunk 1 above)
Comment (oyvindronningstad) [reply samples/bootloader/prj.conf]: `FW_INFO` is no longer `selected` in the bootloader so this makes sense now :)

**Comment by hakonfam** [comment]
@oyvindronningstad  please rebase

**Comment by oyvindronningstad** [comment]
Ready to merge

**Pull request merged**